qt/python/mantidqt/widgets/superplot/presenter.py

```diff
@@ -53,6 +58,30 @@ def __init__(self, canvas, parent=None):
         self._update_spectrum_slider()
         self._update_hold_button()
 
+    def set_bin_mode(self, state):
+        """
+        Set the plot mode.
+
+        Args:
+            state (bool): if true, bin mode is set, if false, spectrum mode
+        """
+        if state:
+            self._view.set_mode(self.BIN_MODE_TEXT)
+        else:
+            self._view.set_mode(self.SPECTRUM_MODE_TEXT)
```

Comment:
This if needs to move to the model as per our [GUI standards](https://developer.mantidproject.org/Standards/GUIStandards.html#no-matter-how-small-logic-goes-into-the-model), and have a test associated.

It's a simple if statement, so it's a pain, but we agreed to do this for all logic as it tends to grow 

Something like this would be fine:
```
self._view.set_mode(self.model.getSpectrumMode(state))
```

Reply:
I am not sure to understand which logic should go in the model. Here, this is only a setter for the state of the view when the superplot is started (i.e. the selected item of the combobox).

Reply:
Sorry with fresh eyes I've just seen its `BIN_MODE_TEXT` rather than `BIN_MODE` which I originally read it as.
Feel free to ignore this, as I assumed it was switching modes rather than text displays